services/inc/ringbuffer.h

```diff
@@ -310,7 +310,7 @@ inline ssize_t RingBuffer<T>::acquireCommit(size_t size, size_t cancel) {
 
     headPending_ -= (size + cancel);
     head_ = wrap(head_ + size, curSize_);
-    full_ = ((head_ == tail_) && size > 0);
+    full_ = ((head_ == tail_) && size > 0) || full_;
```

Comment:
Nit: let's change this for readability:
```c++
if (size > 0) {
    full_ = (head_ == tail_);
}
```

We should only be updating `full_` flag if we are modifying `head_`.